src/client/components/OrdersSummary/OrdersSummary.component.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import OrderComponent from './OrderComponent';
+import './OrdersSummary.styles.css';
+
+export default function OrdersSummary({ orders, orderDetails }) {
```

Comment:
This is fairly inefficient. I imagine it's done this way because it's just for Storybook in this PR and you might not know yet where to put the data or call the API but this code won't perform very well.
Let's see how we can make it better by understanding what it does poorly.

On each render, you will loop over `orders`, and each time you render one `OrderComponent` you will filter the list of `orderDetails` to find the relevant ones. 
Imagine you have 100 orders, and each one has 10 items, `orderDetails` will be 1000 items. Now, that isn't so much but say that your app re-renders often? On top of this, what if the user never clicks to view the details of an order? We're doing all this work for no reason.
Further, why does the single `OrderComponent` decide if the page is in detail mode or not? 
I think it would be better if the page took care of that, so you never end up with bugs where multiple order details are shown at the same time. It also has the benefit of splitting `OrderComponent` into two smaller simpler components.

Imagine instead, if you had an interface like this:
```js
function OrderSummaries({ orders, currentOrderDetails, setCurrentOrder }) {
  return (
    <div className="orders-view">
        <ul className="order-list">
          {orders.map((order) => (
            <li className="order-item">
              <OrderSummary order={order} onClick={() =>setCurrentOrder(order) } />
            </li>
          ))}
        </ul>
       { currentOrderDetails && 
          <OrderDetailsModal details={currentOrderDetails} />
       }
    </div>
  );
}
```

No more looping on each render, no risk of bugs that multiple modals are open, each component is more simple than before.
But where does the logic live then? In the rest of the app, like the Page container.

```js
function OrderSummariesPage({ user_id }) {
  const [currentOrder, setCurrentOrder] = useState();
  const orders = useOrders(user_id);
  const orderDetails = useOrderDetails(currentOrder);

  return (
    <OrderSummaries orders={orders} currentOrderDetails={orderDetails} setCurrentOrder={setCurrentOrder} />
  );
}
```

The `useOrders` hook just calls the API `api/orders?user=user_id` to get a list of the orders, but not the details. `useOrderDetails` fetches the details for the order that is currently being viewed. Since both of these likely can use `useEffect` internally they will update when either the `user_id` changes or the `currentOrder` changes.

The `OrderSummaries` drives that change by showing a list of orders, when an order is clicked, it uses the prop callback `setCurrentOrder` to change the state of the page which causes the page to re-render and call the hooks again to fetch the data. Once the data comes back from the API, `OrderSummaries` gets re-rendered, now with the details data for the order that was click which it can then display.